api/v1alpha1/nutanix_node_types.go

```diff
@@ -0,0 +1,135 @@
+	Image NutanixResourceIdentifier `json:"image"`
+
+	// cluster is to identify the cluster (the Prism Element under management
+	// of the Prism Central), in which the Machine's VM will be created.
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// subnet is to identify the cluster's network subnet to use for the Machine's VM
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	Subnets NutanixResourceIdentifiers `json:"subnet"`
+
+	// Defines the boot type of the virtual machine. Only supports UEFI and Legacy
+	BootType NutanixBootType `json:"bootType,omitempty"`
+
+	// systemDiskSize is size (in Quantity format) of the system disk of the VM
+	// The minimum systemDiskSize is 20Gi bytes
+	SystemDiskSize string `json:"systemDiskSize"`
+}
+
+func (NutanixNodeSpec) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Nutanix Node configuration",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"vcpusPerSocket": {
+					Description: "vcpusPerSocket is the number of vCPUs per socket of the VM",
+					Type:        "integer",
+				},
+				"vcpuSockets": {
+					Description: "vcpuSockets is the number of vCPU sockets of the VM",
+					Type:        "integer",
+				},
+				"memorySize": {
+					Description: "memorySize is the memory size (in Quantity format) of the VM eg. 4Gi",
+					Type:        "string",
+				},
+				"image":    NutanixResourceIdentifier{}.VariableSchema().OpenAPIV3Schema,
+				"cluster":  NutanixResourceIdentifier{}.VariableSchema().OpenAPIV3Schema,
+				"subnet":   NutanixResourceIdentifiers{}.VariableSchema().OpenAPIV3Schema,
+				"bootType": NutanixBootType("legacy").VariableSchema().OpenAPIV3Schema,
+				"systemDiskSize": {
+					Description: "systemDiskSize is size (in Quantity format) of the system disk of the VM eg. 20Gi",
+					Type:        "string",
+				},
+			},
+		},
+	}
+}
+
+// NutanixIdentifierType is an enumeration of different resource identifier types.
+type NutanixIdentifierType capxv1.NutanixIdentifierType
+
+func (NutanixIdentifierType) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type:        "string",
+			Description: "NutanixIdentifierType is an enumeration of different resource identifier types",
+		},
+	}
+}
+
+// NutanixBootType is an enumeration of different boot types.
+type NutanixBootType capxv1.NutanixBootType
+
+func (NutanixBootType) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type:        "string",
+			Description: "NutanixBootType is an enumeration of different boot types.",
```

Comment:
lets add following enum too. was working on in other PR 
<pre>
			Enum: variables.MustMarshalValuesToEnumJSON(
				NutanixBootTypeLegacy,
				NutanixBootTypeUEFI,
			),
</pre>

Reply:
Done in https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/13/commits/ea4005528378751a1d5b24090020a353a8789bb1